File: rb/features.rb
# WorldWonders SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module WorldWondersFeatures
  def self.make_feature(name)
    case name
    when "base"
      WorldWondersBaseFeature.new
    when "ratelimit"
      WorldWondersRatelimitFeature.new
    when "retry"
      WorldWondersRetryFeature.new
    when "test"
      WorldWondersTestFeature.new
    when "timeout"
      WorldWondersTimeoutFeature.new
    else
      WorldWondersBaseFeature.new
    end
  end
end

File: py/features.py
# WorldWonders SDK feature factory

from feature.base_feature import WorldWondersBaseFeature
from feature.test_feature import WorldWondersTestFeature


def _make_feature(name):
    features = {
        "base": lambda: WorldWondersBaseFeature(),
        "test": lambda: WorldWondersTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
